### PR TITLE
Pass return url into verification email from JS button

### PR DESCRIPTION
**Pull request by Calum-Campbell**
## What does this change?
 - Ensures we pass the returnUrl into the verify email link sent from the button on the /verify-email page. 
This is the JS button part of https://github.com/guardian/identity/pull/913

## What is the value of this and can you measure success?
- Ensures people end up where they were originally going when hitting the profil.theguardian pages.

## Tested in CODE?
No

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/api.js [hunk 1]
```diff
@@ -163,13 +164,19 @@ export const getUserEmailSignUps = (): Promise<any> => {
 };
 
 export const sendValidationEmail = (): any => {
+    const returnEndpoint = '/email-prefs';
```
Comment (walaura) [inline static/src/javascripts/projects/common/modules/identity/api.js]: would this be better named `defaultReturnEndpoint ` or `fallbackReturnEndpoint` or something along those lines?

**Review by walaura — APPROVED** [review]
🤘

File: static/src/javascripts/projects/common/modules/identity/api.js [hunk 2]
```diff
@@ -163,13 +164,19 @@ export const getUserEmailSignUps = (): Promise<any> => {
...
+        ? decodeURIComponent(getUrlVars().returnUrl)
+        : (idApiRoot || '') + returnEndpoint;
+
     const request = ajax({
         url: (idApiRoot || '') + endpoint,
         type: 'jsonp',
         crossOrigin: true,
         data: {
             method: 'post',
+            trackingReturnUrl: returnUrl,
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/identity/api.js]: wary of calling this tracking as it's not for tracking.  Do we even use the tracking params once received?

Comment (Calum-Campbell) [reply static/src/javascripts/projects/common/modules/identity/api.js]: To be fair this is the only place I can see the tracking data used in identity https://github.com/guardian/identity/blob/master/identity-api/src/main/scala/com/gu/identity/api/social/SocialUserCreator.scala#L53

**Comment by Calum-Campbell** [comment]
@jfsoul updated to pass returnUrl separately outside of the trackingData

**Pull request merged**